### DISCUSSION TITLE
Rust typed builder

### DISCUSSION THREAD

**Discussion by filnet** [ash-rs/ash#485] [General]
Came across this crate: https://github.com/idanarye/rust-typed-builder.

Interesting ?

**Comment by MarijnS95** [comment]
@filnet Our generated code is already [way too slow](https://github.com/MaikKlein/ash/issues/441#issuecomment-956073795) to compile, and this will add even more burden on the compiler. If anything this should be unrolled (a bit like a preprocessor) at generation time, but it probably doesn't add much if anything that we can't or already generate.
